tests/test_utils.py

```diff
@@ -228,75 +227,85 @@ def test_obfuscate_sensitive_headers(headers, output):
         (
             "http://127.0.0.1",
-            {"NO_PROXY": "127.0.0.1"},
+            {"ALL_PROXY": "http://localhost:123", "NO_PROXY": "127.0.0.1"},
             True,
         ),  # no_proxy as ip case is matched
         (
             "http://127.0.0.1",
-            {"NO_PROXY": "https://127.0.0.1"},
+            {"ALL_PROXY": "http://localhost:123", "NO_PROXY": "https://127.0.0.1"},
             False,
         ),  # no_proxy with scheme is ignored
         (
             "http://127.0.0.1",
-            {"NO_PROXY": "1.1.1.1"},
+            {"ALL_PROXY": "http://localhost:123", "NO_PROXY": "1.1.1.1"},
             False,
         ),  # different no_proxy means its proxied
         (
             "http://courses.mit.edu",
-            {"NO_PROXY": "mit.edu"},
+            {"ALL_PROXY": "http://localhost:123", "NO_PROXY": "mit.edu"},
             True,
         ),  # no_proxy for sub-domain matches
         (
             "https://mit.edu.info",
-            {"NO_PROXY": "mit.edu"},
+            {"ALL_PROXY": "http://localhost:123", "NO_PROXY": "mit.edu"},
             False,
         ),  # domain is actually edu.info, so should be proxied
         (
             "https://mit.edu.info",
-            {"NO_PROXY": "mit.edu,edu.info"},
+            {"ALL_PROXY": "http://localhost:123", "NO_PROXY": "mit.edu,edu.info"},
             True,
         ),  # list in no_proxy, matches second domain
         (
             "https://mit.edu.info",
-            {"NO_PROXY": "mit.edu, edu.info"},
+            {"ALL_PROXY": "http://localhost:123", "NO_PROXY": "mit.edu, edu.info"},
             True,
         ),  # list with spaces in no_proxy
         (
             "https://mit.edu.info",
-            {"NO_PROXY": "mit.edu,mit.info"},
+            {"ALL_PROXY": "http://localhost:123", "NO_PROXY": "mit.edu,mit.info"},
             False,
         ),  # list in no_proxy, without any domain matching
         (
             "https://foo.example.com",
-            {"NO_PROXY": "www.example.com"},
+            {"ALL_PROXY": "http://localhost:123", "NO_PROXY": "www.example.com"},
             False,
         ),  # different subdomains foo vs www means we still proxy
         (
             "https://www.example1.com",
-            {"NO_PROXY": ".example1.com"},
+            {"ALL_PROXY": "http://localhost:123", "NO_PROXY": ".example1.com"},
             True,
         ),  # no_proxy starting with dot
         (
             "https://www.example2.com",
-            {"NO_PROXY": "ample2.com"},
+            {"ALL_PROXY": "http://localhost:123", "NO_PROXY": "ample2.com"},
             False,
         ),  # whole-domain matching
         (
             "https://www.example3.com",
-            {"NO_PROXY": "*"},
+            {"ALL_PROXY": "http://localhost:123", "NO_PROXY": "*"},
             True,
         ),  # wildcard * means nothing proxied
     ],
 )
 def test_should_not_be_proxied(url, no_proxy, expected):
```

Comment:
Should we move this test to `test_proxies.py` now that the `should_not_be_proxied` util is gone?

Reply:
Good call

Reply:
I've merged these tests into the existing `test_proxies_environ` case now.
Looks much nicer!